File: Day4/ProductCrudeDemo2/src/com/demo/service/ProductService.java
package com.demo.service;

import java.util.List;

import com.demo.beans.Product;

public interface ProductService {

	List<Product> getAllProducts();

	boolean addProduct(Product p);

	Product getById(int pid);

	boolean updateProduct(Product p);

	boolean deleteById(int pid);

}

File: Day16/productcrudaxios/src/service/ProductService.js
import axios from 'axios';
let baseUrl="http://localhost:9090/SpringRESTWebService/product/"
class ProductService{
    getAllproducts(){
       return axios.get(baseUrl+"products")
    }
    getById(pid){
        return axios.get(baseUrl+"products/"+pid)
    }
    addProduct(prod){
        return axios.post(baseUrl+"products/"+prod.pid,prod)
    }
    updateProduct(prod){
        return axios.put(baseUrl+"products/"+prod.pid,prod)
    }
    deleteProduct(pid){
        return axios.delete(baseUrl+"products/"+pid);
    }
}

export default new ProductService();